src/xccl/ProcessGroupXCCL.hpp

```diff
@@ -105,7 +171,44 @@ class TORCH_API ProcessGroupXCCL : public Backend {
     friend class ProcessGroupXCCL;
   };
 
-  ProcessGroupXCCL(const c10::intrusive_ptr<Store>& store, int rank, int size);
+  struct Options : public Backend::Options {
+    explicit Options();
+
+    static c10::intrusive_ptr<Options> create() {
+      return c10::make_intrusive<Options>();
+    }
+
+    std::vector<uint64_t> global_ranks_in_group;
+    std::string group_name;
+  };
+
+  class HeartbeatMonitor {
```

Comment:
Do you think if it could be separated in a new file instead of binding with XCCL process? Put  all codes in `ProcessGroupXCCL` to make files too large and hard to maintain in future.